### PR TITLE
Move `system.local` and `system.peers` handling into an interface

### PR DESCRIPTION
**Pull request by mpenick**
Also, fix cases where non-stargate nodes could end up in `system.peers`

### CODE REVIEW

**Comment by mpenick** [comment]
Also, merges other changes for the previous repo.

**Review by EricBorczuk — APPROVED** [review]

**Review by dougwettlaufer — APPROVED** [review]

File: persistence-cassandra-4.0/src/main/java/io/stargate/db/cassandra/impl/CassandraPersistence.java [hunk 1]
```diff
@@ -80,7 +86,7 @@ public void initialize(Config config)
         logger.info("Initializing CassandraPersistence");
         System.setProperty("cassandra.join_ring", "false");
         daemon = new CassandraDaemon(true);
-
```
Comment (dougwettlaufer) [inline persistence-cassandra-4.0/src/main/java/io/stargate/db/cassandra/impl/CassandraPersistence.java:83]: Is this missing `stargate.developer_mode`?

File: persistence-dse-6.8/src/main/java/io/stargate/db/dse/impl/DsePersistence.java [hunk 2]
```diff
@@ -115,15 +113,21 @@ public void initialize(Config config)
             throw new RuntimeException("Unable to start DSE persistence layer", t);
         }
 
-        StargateSystemKeyspace.initialize();
```
Comment (dougwettlaufer) [inline persistence-dse-6.8/src/main/java/io/stargate/db/dse/impl/DsePersistence.java:118]: Is this missing `stargate.developer_mode`?

Comment (mpenick) [reply persistence-dse-6.8/src/main/java/io/stargate/db/dse/impl/DsePersistence.java:118]: It was never added to these branches?

File: persistence-cassandra-4.0/src/main/java/io/stargate/db/cassandra/impl/CassandraPersistence.java (hunk 1 above)
Comment (mpenick) [reply persistence-cassandra-4.0/src/main/java/io/stargate/db/cassandra/impl/CassandraPersistence.java:83]: Yeah, developer mode was never added to these persistence backends.

**Review by dougwettlaufer — APPROVED** [review]

**Comment by mpenick** [comment]
I fixed issues suggested by @jorgebay in https://github.com/stargate/stargate/pull/41

**Review by dougwettlaufer — APPROVED** [review]

**Pull request merged**